opal/corelib/file.rb

```diff
@@ -7,20 +7,28 @@ class File < IO
 
   class << self
     def expand_path(path, basedir = nil)
-      path = [basedir, path].compact.join(SEPARATOR)
-      parts = path.split(SEPARATOR)
+      sep = SEPARATOR
+      path = [basedir, path].compact.join(sep) unless path.start_with?(sep)
+      parts = path.split(sep)
+      absolute = path.start_with?(sep)
       new_parts = []
-      parts[0] = Dir.home if parts.first == '~'
-      parts[0] = Dir.pwd if parts.first == '.'
+
+      unless absolute
+        if parts.first == '~'
+          new_parts << Dir.home
+        else
+          new_parts << Dir.pwd
```

Comment:
As far as I understand C code, I think the `Dir.pwd` is only prepend if the `basedir` argument is `nil`: https://github.com/ruby/ruby/blob/0471781b2a7a40681ccdc051f0239eb19719d79a/file.c#L3414-L3425

And I think `basedir` can be a non absolute path... ?

Reply:
Thanks! I jotted down the code last knight without actually checking the original C 😄

Reply:
> ... without actually checking the original C 

"Great minds think alike." :grinning: 

> I jotted down the code last knight...

"last knight" ?! :horse: :crossed_swords: I think you need some sleep :stuck_out_tongue_winking_eye:

Reply:
> "last knight" ?! 🐴 ⚔️ I think you need some sleep 😜

definitely! 😆